### PR TITLE
feat(ST-134): add get submitted assignments api endpoint

### PR DESCRIPTION
**Pull request by faysal-alam-sazim**
## Description of Change

- Added GET `classrooms/:classroomId/assignments/studentSubmissions` api endpoint inside assignments controller
- Added necessary service functions

## Associated Ticket Link(s)

- https://trello.com/c/qbu9w5is

## Related Pull Request(s)

- N/A

### CODE REVIEW

File: src/assignments/assignments.controller.ts [hunk 1]
```diff
@@ -33,6 +35,16 @@ export class AssignmentsController {
     return this.assignmentsService.getAssignments(classroomId);
   }
 
+  @UseGuards(RolesGuard)
+  @Roles(EUserRole.STUDENT)
+  @Get(":classroomId/assignments/studentSubmissions")
```
Comment (ishmam-iqbal-sazim) [inline src/assignments/assignments.controller.ts:40]:
```suggestion
  @Get(":classroomId/assignments/:assignmentId/submissions")
```

File: src/assignments/assignments.controller.ts [hunk 2]
```diff
@@ -33,6 +35,16 @@ export class AssignmentsController {
...
+  getStudentsSubmission(
+    @Param("classroomId", ParseIntPipe) classroomId: number,
+    @CurrentUser() user: ITokenizedUser,
+  ) {
+    return this.assignmentsService.getStudentSubmittedAssignments(classroomId, user.id);
+  }
+
```
Comment (ishmam-iqbal-sazim) [inline src/assignments/assignments.controller.ts:47]: Shouldn't this be inside the submissions controller?

File: src/assignments/assignments.service.ts [hunk 3]
```diff
@@ -25,6 +29,22 @@ export class AssignmentsService {
...
+
+    const submissions = await this.assignmentSubmissionsRepository.find({ student: student.id });
+
+    const submittedAssignmentIds = submissions.map((submission) => submission.assignment.id);
+
+    const submittedAssignments = assignments.filter((assignment) =>
+      submittedAssignmentIds.includes(assignment.id),
+    );
+
+    return submittedAssignments;
```
Comment (ishmam-iqbal-sazim) [inline src/assignments/assignments.service.ts:45]: Isn't this finding all submissions for all assignments in the classroom?

File: src/assignments/assignments.service.ts [hunk 4]
```diff
@@ -25,6 +29,22 @@ export class AssignmentsService {
     return assignments;
   }
 
+  async getStudentSubmittedAssignments(classroomId: number, userId: number) {
```
Comment (ishmam-iqbal-sazim) [inline src/assignments/assignments.service.ts:32]: This should be inside the submissions service

File: src/assignments/assignments.service.ts (hunk 3 above)
Comment (faysal-alam-sazim) [reply src/assignments/assignments.service.ts:45]: No bhai, it is returning the assignments which user(student) has submitted

File: src/assignments/assignments.controller.ts (hunk 2 above)
Comment (faysal-alam-sazim) [reply src/assignments/assignments.controller.ts:47]: it is returning the assignments not submissions. This is why I put this inside assignment controller

File: src/assignments/assignments.controller.ts (hunk 1 above)
Comment (faysal-alam-sazim) [reply src/assignments/assignments.controller.ts:40]: This route will not return submissions. Maybe I should use `:classroomId/submittedAssignments`

File: src/assignments/assignments.service.ts (hunk 3 above)
Comment (ishmam-iqbal-sazim) [reply src/assignments/assignments.service.ts:45]: Why do you need this?

Comment (faysal-alam-sazim) [reply src/assignments/assignments.service.ts:45]: To disable assignment submit button for student

Comment (ishmam-iqbal-sazim) [reply src/assignments/assignments.service.ts:45]: Do you need to make a separate Api call?

Don't you have access to all the submissions inside the assignment object `assignment.submissions`?

Comment (faysal-alam-sazim) [reply src/assignments/assignments.service.ts:45]: I am going to close this pull request as it is unnecessary,

**Pull request closed without merging**